lib/rucio/core/rule.py

```diff
@@ -2611,21 +2636,26 @@ def __evaluate_did_attach(eval_did, session=None, logger=logging.log):
                                 session.query(models.DatasetLock).filter_by(rule_id=rule.id).update({'state': LockState.STUCK})
                             continue
 
-                        # 3. Apply the Replication rule to the Files
                         preferred_rse_ids = []
-                        # 3.1 Check if the dids in question are files added to a dataset with DATASET/ALL grouping
-                        if new_child_dids[0].child_type == DIDType.FILE and rule.grouping != RuleGrouping.NONE:
-                            # Are there any existing did's in this dataset
-                            brother_did = session.query(models.DataIdentifierAssociation).filter(
-                                models.DataIdentifierAssociation.scope == eval_did.scope,
-                                models.DataIdentifierAssociation.name == eval_did.name).order_by(models.DataIdentifierAssociation.created_at).first()
-                            if brother_did is not None:
-                                # There are other files in the dataset
-                                brother_locks = rucio.core.lock.get_replica_locks(scope=brother_did.child_scope,
-                                                                                  name=brother_did.child_name,
-                                                                                  nowait=True,
-                                                                                  session=session)
-                                preferred_rse_ids = [lock['rse_id'] for lock in brother_locks if lock['rse_id'] in [rse['id'] for rse in rses] and lock['rule_id'] == rule.id]
+                        brother_scope_name = None
```

Comment:
Please add back the comment
```
# 3. Apply the Replication rule to the Files
```
Otherwise all the other numbering doesn't make any sense.